substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -94,17 +94,246 @@ private PosixJavaNetSubstitutions() {
 @TargetClass(className = "java.net.PlainDatagramSocketImpl")
 @Platforms({Platform.LINUX.class, Platform.DARWIN.class})
 final class Target_java_net_PlainDatagramSocketImpl {
+    @Alias boolean connected;
 
     @Substitute
     @SuppressWarnings({"static-method", "unused"})
     protected void bind0(int lport, InetAddress laddr) throws SocketException {
-        throw VMError.unsupportedFeature("Unimplemented: java.net.PlainDatagramSocketImpl.bind0(int, InetAddress)");
+        //        jobject fdObj = (*env)->GetObjectField(env, this, pdsi_fdID);
```

Comment:
You are starting to diverge from the C code.  That makes me have to think!  In code like this I would rather not have to think.  You have the code at line 109, just move it up here and I do not have to wonder what happened to it.  Or wonder how it got to line 109 when I get there.